python_agent/custom_env.py

```
@@ -0,0 +1,448 @@
+        if nearby_npcs is None:
+            print("Attack: No NPC info in last_observation.")
+            return None, {}, {"attack_attempted": False, "error": "NPC info missing"}
+
+        # --- Player Animation for combat state ---
+        # self.player_is_in_combat_animation should be updated in step() method after new obs.
+        # For now, we'll simplify and not use it directly to gate re-attacking.
+
+        best_target_goblin_id = None
+        min_dist_sq = float('inf') # Use squared distance to avoid sqrt
+        
+        player_x, player_y, player_plane = player_loc_data[0], player_loc_data[1], player_loc_data[2]
```

Ruff (0.11.9)

Local variable `player_plane` is assigned to but never used

Remove assignment to unused variable `player_plane`

(F841)